scio-avro/src/main/scala/com/spotify/scio/avro/syntax/ScioContextSyntax.scala

```diff
@@ -43,9 +49,49 @@ final class ScioContextOps(private val self: ScioContext) extends AnyVal {
   def objectFile[T: Coder](path: String): SCollection[T] =
     self.read(ObjectFileIO[T](path))
 
+  /**
+   * Same as [[objectFile]] but reads from a multi-file input. The way how files are read
+   * depends on the hintMatchesManyFiles parameter.
+   *
+   * If the number of files is large (e.g. tens of thousands or more), set hintMatchesManyFiles
+   * to true for better performance and scalability. Note that it may decrease performance if the
+   * number of files is small.
+   */
+  def objectFiles[T : Coder](paths: List[String], hintMatchesManyFiles: Boolean = false)
+  : SCollection[T] = {
+    val coder = CoderMaterializer.beamWithDefault(Coder[T])
+
+    self.readFiles(
+      objectFile[T],
+      new MultiFilePTransform[T] {
+        override def expand(input: PCollection[FileIO.ReadableFile]): PCollection[T] = {
+          input
+            .apply(beam.AvroIO.readFilesGenericRecords(AvroBytesUtil.schema))
+            .apply(ParDo.of(Functions.mapFn[GenericRecord, T](r => AvroBytesUtil.decode(coder, r))))
+        }
+      }
+    )(paths, hintMatchesManyFiles)
+  }
+
   def avroFile(path: String, schema: Schema): SCollection[GenericRecord] =
     self.read(GenericRecordIO(path, schema))
 
+  /**
+   * Same as [[avroFile]] but reads from a multi-file input. The way how files are read depends on
+   * the hintMatchesManyFiles parameter.
+   *
+   * If the number of files is large (e.g. tens of thousands or more), set hintMatchesManyFiles
+   * to true for better performance and scalability. Note that it may decrease performance if the
+   * number of files is small.
+   */
+//  def avroFiles(paths: List[String],
```

Comment:
had to comment this out for now bc it conflicts the other `avroFiles` overload. They both have default args. I guess the only way to get around this would be to give them different names?